apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-nux/src/sharing-modal/suggested-tags.tsx

```diff
@@ -0,0 +1,120 @@
+import { recordTracksEvent } from '@automattic/calypso-analytics';
+import { Button, FormTokenField } from '@wordpress/components';
+import { TokenItem } from '@wordpress/components/build-types/form-token-field/types';
+import { useDispatch, useSelect } from '@wordpress/data';
+import { useEffect } from '@wordpress/element';
+import { store as noticesStore } from '@wordpress/notices';
+import { useI18n } from '@wordpress/react-i18n';
+import * as React from 'react';
+import useAddTagsToPost from './use-add-tags-to-post';
+
+type PostMeta = {
+	reader_suggested_tags: string;
+};
+
+type CoreEditorPlaceholder = {
+	getCurrentPost: ( ...args: unknown[] ) => {
+		id: number;
+		meta: PostMeta;
+	};
+};
+
+type SuggestedTagsEventProps = {
+	number_of_original_suggested_tags: number;
+	number_of_selected_tags: number;
+	number_of_suggested_tags_selected: number;
+	number_of_added_tags: number;
+};
+
+type SuggestedTagsProps = {
+	setShouldShowSuggestedTags: ( shouldShow: boolean ) => void;
+};
+
+function SuggestedTags( props: SuggestedTagsProps ) {
+	const { __, _n } = useI18n();
+	const { id: postId, meta: postMeta } = useSelect(
+		( select ) => ( select( 'core/editor' ) as CoreEditorPlaceholder ).getCurrentPost(),
+		[]
+	);
+	const { createNotice } = useDispatch( noticesStore );
+	const origSuggestedTags = postMeta?.reader_suggested_tags
+		? JSON.parse( postMeta.reader_suggested_tags )
+		: [];
+	const [ selectedTags, setSelectedTags ] = React.useState( origSuggestedTags );
+	const onAddTagsButtonClick = ( numAddedTags: number ) => {
+		// Compare origSuggestedTags and selectedTags and determine the number of tags that are different
+		const numSuggestedTags = origSuggestedTags.length;
+		const numSelectedTags = selectedTags.length;
+		const numSameTags = origSuggestedTags.filter( ( tag: string ) =>
+			selectedTags.includes( tag )
+		).length;
+		const eventProps: SuggestedTagsEventProps = {
+			number_of_original_suggested_tags: numSuggestedTags,
+			number_of_selected_tags: numSelectedTags,
+			number_of_suggested_tags_selected: numSameTags,
+			number_of_added_tags: numAddedTags,
+		};
+		recordTracksEvent( 'calypso_reader_post_publish_add_tags', eventProps );
+		if ( numAddedTags > 0 ) {
+			createNotice(
+				'success',
+				_n( 'Tag Added.', 'Tags Added.', numAddedTags, 'full-site-editing' ),
+				{
+					type: 'snackbar',
+				}
+			);
+		} else {
+			createNotice( 'warning', __( 'No Tags Added.', 'full-site-editing' ), {
+				type: 'snackbar',
+			} );
+		}
+		props.setShouldShowSuggestedTags( false );
+	};
+	const { saveTags } = useAddTagsToPost( postId, selectedTags, onAddTagsButtonClick );
+
+	useEffect( () => {
+		if ( origSuggestedTags?.length === 0 ) {
+			props.setShouldShowSuggestedTags( false );
```

Comment:
DO you think its worth having another tracking event here to signify that we opened this and had no tags to suggest? Or is that redundant with the backend logstash items you setup?

Reply:
I was in 2 minds about it. I was thinking we could figure that out but might be simpler to add it.

Reply:
Ok added new track event that will fire if no suggested tags are found and the site is on a English locale. It should be interesting to track to see if we have many sites that are falling into this category - it would suggest that the post content is too small to be included or perhaps something else that we need to investigate.